### ISSUE TITLE
Unexpected error on `ctypes.POINTER()` array creation 

### ISSUE THREAD

**Issue by plafosse** [microsoft/pyright#1994]
```python
import ctypes

bar = ctypes.c_bool * 3
foo = ctypes.POINTER(ctypes.c_bool) * 3
```

**Describe the bug**

The following error occurs on the line `foo = ctypes.POINTER(Foo) * 3`  which is the standard way of creating arrays in ctypes, it doesn't have this problem with non-POINTER ctypes though.

```
Operator "*" not supported for types "Type[Type[pointer[c_bool]]]" and "Literal[3]"
  Operator "*" not supported for types "Type[Type[pointer[c_bool]]]" and "Literal[3]"
```

**Expected behavior**
Expect no error on this line.

**VS Code extension or command-line**
Latest VSCode extension and tested in Pyright 1.1.149
MacOS 11.2.3

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.150, which I just published. It will also be included in the next release of pylance.
